### PR TITLE
Fix "size" field in the body of AbstractBulkByScrollRequest

### LINKED ISSUE #35636: delete_by_query API "size" parameter not supported in the json request in 6.x
**Elasticsearch version** (`bin/elasticsearch --version`):6.x

**Description of the problem including expected versus actual behavior**:
**Steps to reproduce**:

testing the "size" parameter inside the json request
```
post twitter/doc/1
{
 "message":"hello1",
 "num":2
}
post twitter/doc/2
{
 "message":"hello2",
 "num":1
}
post twitter/doc/3
{
 "message":"hello",
 "num":2
}
post twitter/doc/4
{
 "message":"hello",
 "num":2
}

get twitter/_search

POST twitter/_delete_by_query
{
 "query": {
   "bool": {
     "must": [
       {
         "script": {
           "script": {
             "lang": "expression",
             "source": """doc["num"].value ==2"""
           }
         }
       }
     ]
   }
 }
 , "size":2
}
```
The get twitter/_search returns 2 docs after the delete_by_query call was executed in ES 5.6.x versions.
But it will return 1 doc after the call in 6.x (random testing on 6.0.0, 6.4, 6.5 ), it seems that the "size" param is ignored in the call, the delete_by_query will remove 3 docs instead of 2.

If we add  "size" as a query string param then the delete_by_query will delete 2 docs on all versions tested.
```
POST twitter/_delete_by_query?size=2
{
 "query": {
   "bool": {
     "must": [
       {
         "script": {
           "script": {
             "lang": "expression",
             "source": """doc["num"].value ==2"""
           }
         }
       }
     ]
   }
 }
}
```
Has this option been removed starting 6.x?
Was not able to find any of these changes in the es docs as well.

### PR DESCRIPTION
**Pull request by PnPie**
The "size" field in the body of `AbstractBulkByScrollRequest(DeleteByQuery/UpdateByQuery)` is not taken into consideration anymore during some previous changes, and currently only passing "size" as parameter works for `DeleteByQuery` or `UpdateByQuery`. This is to fix it and make it worked again.

Close #35636

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/reindex/AbstractBulkByScrollRequest.java [hunk 1]
```diff
@@ -130,7 +130,6 @@ public AbstractBulkByScrollRequest(SearchRequest searchRequest, boolean setDefau
         if (setDefaults) {
             searchRequest.scroll(DEFAULT_SCROLL_TIMEOUT);
             searchRequest.source(new SearchSourceBuilder());
-            searchRequest.source().size(DEFAULT_SCROLL_SIZE);
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/index/reindex/AbstractBulkByScrollRequest.java]: I *think* this should stay as it was or else transport client users will start to get a default scroll size of `10` again which won't perform well..

**Review by nik9000 — CHANGES_REQUESTED** [review]
I think this is *almost* right, but it has an issue for transport client users. I think. The trouble is that we didn't have any real testing of the default transport client behavior. But I'm fairly sure we should keep that default size setting. And that means changing Abstract rest controlled slightly. But I think this is on the right track!

**Comment by PnPie** [comment]
Hi @nik9000 I changed [search XContent body parsing](https://github.com/elastic/elasticsearch/pull/35742/files#diff-373ad6c00a31d289415cc037657acb7bR1006) part and passed a setSize consumer inside to be able to "consume" `size` field in search request's body.
Otherwise it seems a little bit tricky on Abstract rest level to idenfity whether the `size` field has been set or not ([like this](https://github.com/elastic/elasticsearch/pull/35742/files#diff-de8f3e7c34ac2d22b99602b4b6cefd10R58)), as by default it's 1000.
What do you think of this ?

**Comment by henningandersen** [comment]
@PnPie thanks for your efforts on providing this solution. Unfortunately, we have since implemented a different solution in #41894 where we renamed size to max_docs, fixing confusion on the different size parameters in place. I will therefore close this PR.

**Pull request closed without merging**